### ISSUE TITLE
CPU 100% usage

### ISSUE THREAD

**Issue by chemzqm** [neoclide/coc-highlight#14]
Should be regex related.

**Comment by ashb** [comment]
I've noticed this occasionally too. Anything I can do to help track this down?

**Comment by ashb** [comment]
I just noticed this in my CocInfo output:

```
[Error  - 12:24:53] (node:20361) UnhandledPromiseRejectionWarning: Error: Unhandled method highlight/iskeyword
    at handleResponse (/home/ash/.config/coc/extensions/node_modules/coc-highlight/lib/server.js:614:48)
    at processMessageQueue (/home/ash/.config/coc/extensions/node_modules/coc-highlight/lib/server.js:441:17)
    at Immediate.<anonymous> (/home/ash/.config/coc/extensions/node_modules/coc-highlight/lib/server.js:425:13)
    at processImmediate (internal/timers.js:456:21)
```

**Comment by jc00ke** [comment]
Just ran into this
![image](https://user-images.githubusercontent.com/18191/77486730-bafc6000-6ded-11ea-8662-471f74f6ef43.png)

**Comment by jc00ke** [comment]
I've since switched to https://github.com/ap/vim-css-color will see how that pans out. If it doesn't peg my CPU maybe we could use a few things from it.

**Comment by ungrim97** [comment]
I have seen this issue running on a mac. Let me know if there is any specific info you need

**Comment by phush0** [comment]
I have same problem, uninstalling coc-highlight fix things. I use it with ccls to show read and writes to variables.

**Comment by DiegoGuidaF** [comment]
Same issue here, happens with long files.
Maybe limiting highlighting to files of a maximum length would fix this.

**Comment by GopherJ** [comment]
#48

**Comment by chemzqm** [comment]
Should be fixed on version 2.0.0 which parse line one by one and have cancellation support.
